src/DynamicData/Binding/SortAndBind.cs

```diff
@@ -17,173 +16,233 @@ namespace DynamicData.Binding;
+                             *
+                             *  Use the linear methods as binary search does not work if we do not have an already sorted list.
+                             *  Otherwise, SortAndBindWithBinarySearch.Refresh() unit test will break.
+                             *
+                             * If consumers are using BinarySearch and a refresh event is sent here, they probably should exclude refresh
+                             * events with .WhereReasonsAreNot(ChangeReason.Refresh), but it may be problematic to exclude refresh automatically
+                             * as that would effectively be swallowing an error.
+                             */
+                            var currentIndex = target.IndexOf(item);
+                            var updatedIndex = GetInsertPositionLinear(item);
+
+                            // We need to recalibrate as GetInsertPosition includes the current item
+                            updatedIndex = currentIndex < updatedIndex ? updatedIndex - 1 : updatedIndex;
+                            if (updatedIndex != currentIndex)
+                            {
+                                target.RemoveAt(currentIndex);
+                                target.Insert(updatedIndex, item);
+                            }
+                        }
+                        break;
+                    case ChangeReason.Moved:
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
             }
         }
-    }
-
-    private int GetCurrentPosition(TObject item)
-    {
-        var index = options.UseBinarySearch ? target.BinarySearch(item, comparer) : target.IndexOf(item);
 
-        if (index < 0)
+        private int GetCurrentPosition(TObject item)
         {
-            throw new SortException($"Cannot find item: {typeof(TObject).Name} -> {item} from {target.Count} items");
-        }
+            var index = options.UseBinarySearch ? target.BinarySearch(item, comparer) : target.IndexOf(item);
 
-        return index;
-    }
+            if (index < 0)
+            {
+                throw new SortException($"Cannot find item: {typeof(TObject).Name} -> {item} from {target.Count} items");
+            }
 
-    private int GetInsertPosition(TObject item) => options.UseBinarySearch ? GetInsertPositionBinary(item) : GetInsertPositionLinear(item);
+            return index;
+        }
 
-    private int GetInsertPositionBinary(TObject item)
-    {
-        var index = target.BinarySearch(item, comparer);
-        var insertIndex = ~index;
+        private int GetInsertPosition(TObject item) => options.UseBinarySearch ? GetInsertPositionBinary(item) : GetInsertPositionLinear(item);
 
-        // sort is not returning uniqueness
-        if (insertIndex < 0)
+        private int GetInsertPositionBinary(TObject item)
```

Comment:
Wow, this diff is obnoxiously terrible in GitHub, BTW. The syntax-aware diff in VS is so much better. Never seen a case this bad before.